client/src/pages/Balance/Checkout.jsx

```diff
@@ -0,0 +1,232 @@
+      />
+    </svg>
+    {children}
+  </div>
+);
+
+const stripePromise = loadStripe("pk_test_Y2tTju8cgfUZW4vozmYVG26J00yEwRHw7v");
+
+const Checkout = ({
+  refillAmount,
+  setRefillAmount,
+  setCheckoutPage,
+}) => {
+  const classes = useStyles();
+  const { user } = useContext(UserContext);
+  const stripe = useStripe();
+  const elements = useElements();
+  const [error, setError] = useState(null);
+  const [cardComplete, setCardComplete] = useState(false);
+  const [processing, setProcessing] = useState(false);
+  const [billingDetails, setBillingDetails] = useState({ name: "" });
+  const [paymentSuccessful, setPaymentSuccessful] = useState(false);
+
+  const handleSubmit = async event => {
+    event.preventDefault();
+
+    // Stripe.js has not yet loaded.
+    if (!stripe || !elements) return;
+    if (!billingDetails.name) {
+      document.getElementById("name").focus();
+      setError({ message: "Please enter your name" });
+      return;
+    }
+    if (error || !cardComplete ) {
+      elements.getElement("card").focus();
+      setError({ message: "Invalid card number" });
+      return;
+    }
+    
+    setProcessing(true);
+
+    const requestPaymentIntent = await axios.post(`/api/users/${user.id}/purchase`, { refillAmount });
+
+    const payload = await stripe.createPaymentMethod({
+      type: "card",
+      card: elements.getElement(CardElement),
+      billing_details: billingDetails,
+    });
+
+    const confirm = await stripe.confirmCardPayment(
+      `${requestPaymentIntent.data.clientSecret}`,
+      {
+        payment_method: payload.paymentMethod.id
+      }
+    );
+
+    if (confirm.paymentIntent.status === "succeeded") {
+      setProcessing(false);
+      setPaymentSuccessful(true);
+      user.balance += refillAmount;
+      setRefillAmount(1);
+      await axios.put(`/api/users/${user.id}/add-credits`, { refillAmount });
+    } else if (confirm.error) {
+      setProcessing(false);
+      setError(confirm.error);
+    }
+  };
+
+  return !paymentSuccessful ? (
+    <Grid container direction="column" alignItems="center">
+      <Grid
+        item
+        container
+        xs={12}
+        direction="column"
+        className={classes.stripePayment}
+      >
+        <form className="Form" onSubmit={handleSubmit}>
+          <Grid container direction="column" spacing={2}>
+            <Grid item>
+              <fieldset className="FormGroup">
+                <div className="FormRow">
+                  <input
```

Comment:
We will want to improve styling, here is a specific example where you can use Material-ui textfield